### PR TITLE
Remove importresolver package

### PR DESCRIPTION
**Pull request by pranjalvachaspati-toast**
This replaces enough stuff with kotlinpoet code that all the imports are resolved and tests pass.

### CODE REVIEW

File: protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/DeserializerAnnotator.kt [hunk 1]
```diff
@@ -217,32 +202,59 @@ private constructor(
...
-        packed = packed,
-        options = deserializeOptions(f, ctx)
-    )
+fun deserializeString(f: StandardField, ctx: Context, packed: Boolean): String {
+    val options = deserializeOptions(f, ctx)
+    val read = CodeBlock.of("deserializer.%L", interceptReadFn(f, f.readFn()))
+    val wrappedRead = options?.let { wrapField(it.wrapName, read, it.type, it.oneof) } ?: read.toString()
+    return when {
+        f.map -> deserializeMap(f, options, read)
+        // · is a non-wrapping space
```
Comment (andrewparmet) [inline protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/DeserializerAnnotator.kt:219]: This comment feels a bit unnecessary?

File: protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/DeserializerAnnotator.kt [hunk 2]
```diff
@@ -44,28 +48,34 @@ import com.toasttab.protokt.codegen.protoc.Tag
...
         val properties = annotateProperties(msg, ctx)
 
         return TypeSpec.companionObjectBuilder("Deserializer")
+            // KtDeserializer<MessageType>
             .addSuperinterface(
                 KtDeserializer::class
                     .asTypeName()
                     .parameterizedBy(msg.typeName)
             )
+            // (MessageTypeDsl.() -> Unit) -> Message
```
Comment (andrewparmet) [inline protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/DeserializerAnnotator.kt:78]: Hmm... not sold on the necessity of these comments. Perhaps extracting to a named function would be good?

File: protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/DeserializerAnnotator.kt [hunk 3]
```diff
@@ -125,29 +123,16 @@ private constructor(
                             Unit::class.asTypeName()
                         )
                     )
-                    .addCode("return ${msg.name}Dsl().apply(dsl).build()")
+                    .addStatement("return ${msg.name}Dsl().apply(dsl).build()")
```
Comment (andrewparmet) [inline protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/DeserializerAnnotator.kt:126]: We should be able to use the message's `dslTypeName` here now.

File: protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/DeserializerAnnotator.kt [hunk 4]
```diff
@@ -217,32 +202,59 @@ private constructor(
     )
 
     private fun oneofDes(f: Oneof, ff: StandardField) =
-        OneofTemplate.Deserialize.render(
-            oneof = f.name,
-            name = f.fieldTypeNames.getValue(ff.name),
-            read = deserializeString(ff, ctx, false)
-        )
+        "${f.name}.${f.fieldTypeNames.getValue(ff.fieldName)}(${deserializeString(ff, ctx, false)})"
```
Comment (andrewparmet) [inline protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/DeserializerAnnotator.kt:205]: Yikes. (I know this came straight from StringTemplate.)

File: protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/model/FieldType.kt [hunk 5]
```diff
@@ -95,43 +105,57 @@ private sealed class TypeImpl {
 
 private sealed class Nonscalar(
     override val kotlinRepresentation: KClass<*>? = null,
-    override val ktRepresentation: KClass<*>? = null
+    override val ktRepresentation: KClass<*>? = null,
+    override val isGeneratedType: Boolean = false,
+    override val defaultValue: CodeBlock = CodeBlock.of("null")
```
Comment (andrewparmet) [inline protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/model/FieldType.kt:110]: I think it ended up nice to not bind these models to a particular code generator implementation - can this be implemented with a `when()` statement in the annotator(s) that needs it?

File: protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/DeserializerAnnotator.kt (hunk 4 above)
Comment (pranjalvachaspati-toast) [reply protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/DeserializerAnnotator.kt:203]: Yup, there are a few messy templates in here that should probably be cleaned up and kotlinpoet-ified.

File: protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/MessageSizeAnnotator.kt [hunk 6]
```diff
@@ -157,43 +152,76 @@ private constructor(
...
+                buildCodeBlock {
+                    addNamed(
+                        "%sizeof:M(%tag:T(${f.number})) + " +
+                            "$name.sumOf·{ %sizeof:M(${f.box("it")}) }.let·{ it + %sizeof:M(%uInt32:T(it)) }",
+                        map
+                    )
+                }
+            }
+            f.repeated -> {
+                val map = LinkedHashMap<String, Any>()
```
[the remark this answers is not in the document]
Comment (pranjalvachaspati-toast) [reply protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/MessageSizeAnnotator.kt]: Fixed

File: protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/DslAnnotator.kt [hunk 7]
```diff
@@ -84,23 +87,23 @@ class DslAnnotator(
...
-                                            .addCode("field = copyMap(newValue)")
+                                            .addCode("field = %M(newValue)", runtimeFunction("copyMap"))
                                             .build()
                                     )
                                 } else if (it.repeated) {
                                     setter(
                                         FunSpec.setterBuilder()
                                             .addParameter("newValue", List::class)
-                                            .addCode("field = copyList(newValue)")
+                                            .addCode("field = %M(newValue)", runtimeFunction("finishList"))
```
[the remark this answers is not in the document]
Comment (pranjalvachaspati-toast) [reply protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/DslAnnotator.kt]: Fixed

File: protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/MessageSizeAnnotator.kt [hunk 8]
```diff
@@ -157,43 +152,79 @@ private constructor(
...
+                buildCodeBlock {
+                    addNamed(
+                        "%sizeof:M(%tag:T(${f.number})) + " +
+                            "$name.sumOf·{ %sizeof:M(${f.box("it")}) }.let·{ it + %sizeof:M(%uInt32:T(it)) }",
+                        map
+                    )
+                }
+            }
+            f.repeated -> {
+                val map = mutableMapOf(
```
Comment (andrewparmet) [inline protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/MessageSizeAnnotator.kt:181]: Can these be declared in line as non-mutable maps?

**Pull request merged**